lib/src/facts/linux/networking_resolver.cc

```diff
@@ -142,6 +142,12 @@ namespace facter { namespace facts { namespace linux {
         auto parse_route_line = [&known_route_types](string& line, std::vector<route>& routes) {
             vector<boost::iterator_range<string::iterator>> parts;
             boost::split(parts, line, boost::is_space(), boost::token_compress_on);
+
+            // remove terminating "onlink" flag
+            std::string last_token(parts.back().begin(), parts.back().end());
+            if (last_token == "onlink")
```

Comment:
We should probably also look for "pervasive" (according to the manpage http://www.dsm.fordham.edu/cgi-bin/man-cgi.pl?topic=ip-route&ampsect=8, see the NHFLAGS line)

Other than that, I'm not sure if there are any other peculiarities of parsing the route format. It's probably not worth writing a full-blown parser unless we get more reports of issues with this simple one, though